packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -117,8 +144,14 @@ library TaikoData {
         address assignedProver; // slot 2
         uint96 livenessBond;
         uint64 blockId; // slot 3
-        uint64 proposedAt; // timestamp
-        uint64 proposedIn; // L1 block number, required/used by node/client.
+        // Before the fork, this field is the L1 timestamp when this block is proposed.
+        // After the fork, this is the timestamp of the L2 block.
+        // In a later fork, we an rename this field to `timestamp`.
+        uint64 proposedAt;
+        // Before the fork, this field is the L1 block number where this block is proposed.
+        // After the fork, this is the L1 block number input for the anchor transaction.
+        // In a later fork, we an rename this field to `anchorBlockId`.
+        uint64 proposedIn;
```

Comment:
Same here, we could still use the values like we did previously. Or is this part of the optimization?

Reply:
Moved "proposedAt" and "proposedIn" to the metadata to save gas, but since we need to write "timestamp" and "anchorBlockId" to storage, to avoid writing to a new slot, I simply reused the "proposedAt" and "proposedIn" fields without renaming them. So yes it's optimizaiton.

Reply:
I think no need to write write `timestamp` and `anchorBlockId` to storage? I believe those are only used on L2, not on L1, so just having them as part of the meta hash is sufficient as far as I know.

Reply:
When a new block is proposed, we now read the parent block's `timestamp` and `anchorBlockId` to make sure the new block's values are greater than its parent's. Therefore we need to write these two values into storage.

Reply:
Oh right I also implemented it like that haha. In my mind this was already checked on L2 instead, but all good!